### DISCUSSION TITLE
Four repetitive tasks.

### DISCUSSION THREAD

**Discussion by M-i-c-h-a-l-S** [spakin/SimpInkScr#82] [Q&A] [answered]
I am really glad I found this extension, as I was given dozens of inkscape files to work with. My task is simple, therefore very fitting for Simple Inkscape Scripting!

I need to split one file into eight. What do I do is I select the part I want to save, reverse selection, delete rest, select all and transform='translate(0, 0)' save it, redo all changes and repeat with another part of the pictures.

I've learned from wiki here that some work with selection is possible, however, I have no clue how to revert it.

Thank you for any kind of help!

**Comment by spakin** [comment, marked as answer]
Inspired by your question, I added a few features to Simple Inkscape Scripting that you may find useful: [`unremove`](https://github.com/spakin/SimpInkScr/wiki/Modifying-existing-objects#removing-an-existing-object), which undoes an invocation of `remove`, and [`save_file`](https://github.com/spakin/SimpInkScr/wiki/Other-features#saving-the-current-document-to-a-file), which saves the current document to a file.

For your use case, is the selection something that can be automated, or does it require a human to identify what should be saved?  If the former, I'd recommend _not_ using selections.  Instead, store the unwanted shapes in a list, `remove` each of them, process the remaining objects, and finally `unremove` the shapes in the list.  If the latter, you can iterate over all shapes in `all_shapes`.  `remove` those that are in `selected_shapes` and process those that aren't.  Then `unremove` the list of shapes you removed:
```Python
sel = set(selected_shapes())
for obj in all_shapes():
    if obj in sel:
        # Object is selected.
    else:
        # Object is not selected.
```

Does that help?

**Reply by M-i-c-h-a-l-S** [reply]
You are most kind. Thank you very very much!
